firmware/application/src/app_cmd.c

```diff
@@ -24,59 +24,66 @@
 NRF_LOG_MODULE_REGISTER();
 
 
+static void change_slot_auto(uint8_t slot) {
+    device_mode_t mode = get_device_mode();
+    tag_emulation_change_slot(slot, mode != DEVICE_MODE_READER);
+    light_up_by_slot();
+    set_slot_light_color(0);
+}
+
 
-data_frame_tx_t *cmd_processor_get_version(uint16_t cmd, uint16_t status, uint16_t length, uint8_t *data) {
-    uint16_t version = FW_VER_NUM;
-    return data_frame_make(cmd, status, 2, (uint8_t *)&version);
+static data_frame_tx_t *cmd_processor_get_app_version(uint16_t cmd, uint16_t status, uint16_t length, uint8_t *data) {
+    struct {
+        uint8_t version_major;
+        uint8_t version_minor;
+    } PACKED payload;
+    payload.version_major = APP_FW_VER_MAJOR;
+    payload.version_minor = APP_FW_VER_MINOR;
+    return data_frame_make(cmd, STATUS_DEVICE_SUCCESS, sizeof(payload), (uint8_t *)&payload);
 }
 
 
-data_frame_tx_t *cmd_processor_get_git_version(uint16_t cmd, uint16_t status, uint16_t length, uint8_t *data) {
-    return data_frame_make(cmd, status, strlen(GIT_VERSION), (uint8_t *)GIT_VERSION);
+static data_frame_tx_t *cmd_processor_get_git_version(uint16_t cmd, uint16_t status, uint16_t length, uint8_t *data) {
+    return data_frame_make(cmd, STATUS_DEVICE_SUCCESS, strlen(GIT_VERSION), (uint8_t *)GIT_VERSION);
 }
 
 
-data_frame_tx_t *cmd_processor_get_device(uint16_t cmd, uint16_t status, uint16_t length, uint8_t *data) {
+static data_frame_tx_t *cmd_processor_get_device_model(uint16_t cmd, uint16_t status, uint16_t length, uint8_t *data) {
 #if defined(PROJECT_CHAMELEON_ULTRA)
-    uint8_t device = 1;
+    uint8_t resp_data = 1;
```

Comment:
So...   ULTRA is `1` or `0`?

Reply:
Ooops didn't see that.
It should probably se swapped there for consistency imho.
Let's wait for @doegox opinion on this.

Reply:
In BLE DFU adv data, use 0 to represent `ULTRA`

https://github.com/RfidResearchGroup/ChameleonUltra/blob/98605be5b8cf93bf6e54e8f84dab960dd8879f9a/firmware/nrf52_sdk/components/libraries/bootloader/ble_dfu/nrf_dfu_ble.c#L210

I think we need to change one of both.

Reply:
we can't change `m_device_type` defined in the bootloader, so we'll change the recent command `DATA_CMD_GET_DEVICE_MODEL`